pkg/server/handler.go

```diff
@@ -10,51 +10,69 @@ import (
 	"fmt"
 	"log/slog"
 	"net/http"
+	"reflect"
 
-	"github.com/seatgeek/mailroom/pkg/handler"
+	"github.com/seatgeek/mailroom/pkg/event"
 	"github.com/seatgeek/mailroom/pkg/notifier"
 )
 
-// CreateEventHandler returns a handlerFunc that can be used to handle incoming webhooks
+// CreateEventProcessingHandler returns a handlerFunc that can be used to handle incoming webhooks.
 // It choreographs the parsing of the incoming request, the generation of notifications, dispatching the notifications
 // to the notifier, and returning a success or error response to the client.
-func CreateEventHandler(s handler.Handler, n notifier.Notifier) http.HandlerFunc {
+func CreateEventProcessingHandler(parserKey string, parser event.Parser, processors []event.Processor, ntfr notifier.Notifier) http.HandlerFunc { //nolint:revive
 	return func(writer http.ResponseWriter, request *http.Request) {
-		slog.Debug("handling incoming webhook", "handler", s.Key(), "path", request.URL.Path)
+		slog.Debug("handling incoming webhook", "parser", parserKey, "path", request.URL.Path)
```

Comment:
maybe we should add these as attributes so they can be reused in the logger scoped to this func?

```
logger := slog.With(
    slog.String("parser", parserKey),
    slog.String("path", request.URL.Path),
)
```